### PR TITLE
Source Amplitude - Updates `cursor_field` in the Amplitude Events stream to the proper field

### PR DESCRIPTION
**Pull request by myles-novick**
## What
Addressing Issue #16943 - Source amplitude uses wrong field as "cursor_field"

## How
Updates "cursor_field" in the Amplitude Events stream to the proper field

## Recommended reading order
1. `api.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by sajarin** [comment]
Hey @myles-novick thanks for the PR. This PR is part of Airbyte's Community Maintainer program and will be reviewed by a member of the community. 

@Nmaxime has been assigned to review and help get this merged into master.

**Review by Nmaxime — APPROVED** [review]
LGTM. As explained in the [issue](https://github.com/airbytehq/airbyte/issues/16943) and the API doc, the `server_upload_time` field is the one used by the date range parameters.

**Comment by sajarin** [comment]
Thanks @Nmaxime for the review. 

@bazarnov since you're the DRI for this connector, could I get a final spot check here before we merge?

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py
Comment (bazarnov) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:138]: Could you please give more context about why `server_upload_time` is a better cursor?

**Comment by sajarin** [comment]
@myles-novick can you reply to the comment addressed above?

**Comment by myles-novick** [comment]
> @myles-novick can you reply to the comment addressed above?

sorry, didn't realize that was directed at me

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py
Comment (myles-novick) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:138]: [Amplitude documentation](https://www.docs.developers.amplitude.com/analytics/apis/export-api/#considerations) of the Export API (see: first bullet point under "Considerations") states that the intended cursor field for the Export API is `server_upload_time`:
"The specified date range refers to the time the event data was uploaded to Amplitude servers (see server_upload_time)."

Comment (bazarnov) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:138]: Great finding, thank you.

**Comment by sajarin** [comment]
@myles-novick just making sure you saw the above comment. Would love to get this merged sometime next week!

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py
Comment (myles-novick) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:138]: Can you explain how the code would need to be changed to reflect this? From what I can tell, `stream_state` for an existing connection would look something like `{"event_time": [timestamp]}`, and you're worried that anywhere we read `stream_state.get(self.cursor_field)` or `stream_state[self.cursor_field]`, we will retrieve an undefined value, correct? Are you saying in these places we should be checking both `self.cursor_field` and `"event_time"` explicitly (i.e., hard-coding the string `"event_time"`) in these places?

Sorry, this is my first time contributing to this codebase, I'm not sure I have a complete grasp on the ins and outs yet.

Comment (bazarnov) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:139]: The situation you described is correct, at present. Specifically saying - yes, you need to do additional check for the old cursor value, in methods, where the `stream_state` is initialized as variable or directly called as argument, especially the places where the state compares to other values.

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Comment by marcosmarxm** [comment]
@myles-novick can you check unit test failure/

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by sh4sh** [comment]
Hey @myles-novick , are you interested in continuing with this PR?

**Comment by marcosmarxm** [comment]
Closing due inactivity @myles-novick let me know if you want to continue the work here.

**Pull request closed without merging**
